src/agents/policy_retriever_agent.py

```diff
@@ -0,0 +1,135 @@
+import os
+import sys
+sys.path.append("..")
+
+from langchain_core.prompts import ChatPromptTemplate
+from langchain_community.vectorstores import Chroma
+from openai import OpenAI, AzureOpenAI
+
+import config as cfg
+from helpers.embedding_helpers import OpenAIEmbeddingFunction
+
+# Environment setup
+os.environ['KMP_DUPLICATE_LIB_OK'] = 'True'
+
+# Fuse to blob storage to access chroma db and policy documents
+os.system('sudo blobfuse /home/azureuser/cloudfiles/code/blobfuse/rndaistoragemeldingen --tmp-path=/mnt/resource/blobfusetmp_meldingen --config-file=/home/azureuser/cloudfiles/code/blobfuse/fuse_connection_meldingen.cfg -o attr_timeout=3600 -o entry_timeout=3600 -o negative_timeout=3600 -o allow_other -o nonempty')
+
+class PolicyRetrieverAgent:
+
+    def __init__(self, melding):
+        self.melding = melding
+
+    def retrieve_policy(self):
+        summarized_melding = self.summarize_melding()
+        context_text, sources = self.search_db(summarized_melding)
+        prompt = self.create_prompt(context_text)
+        response_text = self.invoke_model(prompt)
+        formatted_response = self.format_response(response_text, sources)
+        return formatted_response
+
+    def summarize_melding(self):
```

Comment:
Have you checked if this summarization step really impacts performance? Are there examples when the correct documents weren't retrieved without summarization and this extra LLM call fixed it?
It might seem like a small thing, but this solution already has so many LLM calls, I think it's good to be cautious about unnecessary complexity and environmental impact

Reply:
That's maybe nice! I couldn't quickly figure out the argument generation, but is there an easy way to get the central agent to generate that boolean itself? E.g. if a melding is too large?

Reply:
Hmm that might be an option as well. Probably isn't that straigtforward as saying "obove character length _x_ summarize the query", since there might be more reasons why or why not to summarize though...  Maybe for now just do a simple boolean which can be adjusted manually?

Reply:
Cool! Sounds like a great config/settings material to me, so we can easily toggle it per experiment/preference/whatever